autonomy/analyse/service.py

```diff
@@ -168,19 +171,62 @@
         "service_registry_address",
         "on_chain_service_id",
         "share_tm_config_on_startup",
-        "tendermint_p2p_url",
+        "use_termination",
         "setup",
     ],
 }
 
 ABCI = "abci"
 LEDGER = "ledger"
 
-UNKNOWN_LEDGER_RE = re.compile(r".*\(\'(.*)\' was unexpected\)")
-ABCI_CONNECTION_VALIDATOR = Draft4Validator(schema=ABCI_CONNECTION_SCHEMA)
-LEDGER_CONNECTION_VALIDATOR = Draft4Validator(schema=LEDGER_CONNECTION_SCHEMA)
-ABCI_SKILL_VALIDATOR = Draft4Validator(schema=ABCI_SKILL_SCHEMA)
-ABCI_SKILL_MODEL_PARAMS_VALIDATOR = Draft4Validator(
+REQUIRED_PROPERTY_RE = re.compile(r"'(.*)' is a required property")
+PROPERTY_NOT_ALLOWED_RE = re.compile(
+    r"Additional properties are not allowed \((('(.*)'(,)? )+)(was|were) unexpected\)"
+)
+
+
+class CustomSchemaValidator(Draft4Validator):
+    """Custom schema validator to report all missing fields at once."""
+
+    def validate(self, *args: Any, **kwargs: Any) -> None:
+        """Validate and raise all errors at once."""
+        missing = []
+        extra = []
+        for error in self.iter_errors(*args, **kwargs):
```

Comment:
why not create ValidationError with list of all errors strings and show to user as is, without grouping? will make code more simple

Reply:
To do this we will have to create multiple error string for each missing and each additional property which will defeat the purpose of having this class for reporting all missing and additional properties at once

Reply:
@angrybayblade 
we can create custom exception with all errors incapsulated and on cli side print this error message properly line by line

Reply:
I tried to address it here [1d63285](https://github.com/valory-xyz/open-autonomy/pull/1942/commits/1d63285e9626f4df1dad827b66896848a36b420f)